.drone.yml

```diff
@@ -1,168 +1,133 @@
 ---
 kind: pipeline
 type: kubernetes
-name: lint
-
+name: build
 trigger:
   event:
     - push
+  branch:
+    exclude:
+      - main
 
 steps:
-- name: lint bash
-  image: koalaman/shellcheck-alpine
-  commands:
-    - shellcheck *.sh */*.sh */*/*.sh
-  depends_on:
-    - clone
+  - name: build & push
+    image: plugins/docker
+    settings:
+      registry: quay.io
+      repo: quay.io/ukhomeofficedigital/hocs-toolbox
+      tags:
+        - ${DRONE_COMMIT_SHA}
+    environment:
+      DOCKER_PASSWORD:
+        from_secret: QUAY_ROBOT_TOKEN
+      DOCKER_USERNAME: ukhomeofficedigital+hocs_quay_robot
+    depends_on:
+      - clone
 
 ---
 kind: pipeline
 type: kubernetes
-name: build
+name: deploy
+trigger:
+  event:
+    - promote
 
-platform:
-  os: linux
-  arch: amd64
+steps:
+  - name: fetch and checkout
+    image: alpine/git
+    commands:
+      - git fetch --tags
+      - git checkout $${VERSION}
 
+  - name: deploy to prod
+    image: quay.io/ukhomeofficedigital/kd
+    commands:
+      - cd kube
+      - ./deploy.sh
+    environment:
+      ENVIRONMENT: ${DRONE_DEPLOY_TO}
+      KUBE_TOKEN:
+        from_secret: hocs_toolbox_${DRONE_DEPLOY_TO/-/_}
+      KUBE_SERVER: https://kube-api-prod.prod.acp.homeoffice.gov.uk
+    depends_on:
+      - fetch and checkout
+    when:
+      target:
+        include:
+          - "*-prod"
+
+---
+kind: pipeline
+type: kubernetes
+name: build tag
 trigger:
+  event:
+    - tag
   branch:
     - main
-  event:
-    - push
 
 steps:
-  - name: build & push safe
+
+  - name: build & push tag main
     image: plugins/docker
-    depends_on:
-      - clone
     settings:
       registry: quay.io
       repo: quay.io/ukhomeofficedigital/hocs-toolbox
-      target: safe
       tags:
-        - build_${DRONE_BUILD_NUMBER}
-        - ${DRONE_COMMIT_SHA}-safe
+        - ${DRONE_TAG}
+        - ${DRONE_COMMIT_SHA}
         - latest
     environment:
       DOCKER_PASSWORD:
         from_secret: QUAY_ROBOT_TOKEN
       DOCKER_USERNAME: ukhomeofficedigital+hocs_quay_robot
-
-  - name: build & push scary
     depends_on:
-      - build & push safe
-    image: plugins/docker
-    settings:
-      registry: quay.io
-      repo: quay.io/ukhomeofficedigital/hocs-toolbox
-      target: dangerous
-      tags:
-        - ${DRONE_COMMIT_SHA}-dangerous
-    environment:
-      DOCKER_PASSWORD:
-        from_secret: QUAY_ROBOT_TOKEN
-      DOCKER_USERNAME: ukhomeofficedigital+hocs_quay_robot
+      - test project
+    when:
+      branch:
+        - main
 
 ---
-
 kind: pipeline
 type: kubernetes
-name: deploy
-depends_on:
-  - build
-
+name: deploy tag
 trigger:
+  event:
+    - tag
   branch:
     - main
```

Comment:
```suggestion
```
Tags don't have branches.

Reply:
this is an undocumented 'feature' of drone, it breaks if we remove it because it ends up deploying every tag event including on task branches/hotfix branches